test/FluentMigrator.Tests/Containers/FirebirdContainer.cs

```diff
@@ -0,0 +1,33 @@
+﻿#region License
+// Copyright (c) 2024, Fluent Migrator Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using DotNet.Testcontainers.Containers;
+
+using Testcontainers.FirebirdSql;
+
+namespace FluentMigrator.Tests.Containers;
+
+public class FirebirdContainer : ContainerBase
+{
+    /// <inheritdoc />
+    protected override IntegrationTestOptions.DatabaseServerOptions ServerOptions => IntegrationTestOptions.Firebird;
+
+    /// <inheritdoc />
+    protected override int Port => 3050;
+
+    /// <inheritdoc />
+    protected override DockerContainer Build() => new FirebirdSqlBuilder().WithReuse(true).Build();
```

Comment:
I think we should make it easy to do logging.

https://github.com/testcontainers/testcontainers-dotnet/blob/develop/docs%2Fcustom_configuration%2Findex.md 

Nunit supports TestContext.Parameters passed in via --testparam. We can also allow reading a Boolean environment variable for users who always want to turn on container logging: FLUENTMIGRATOR_TESTCONTAINERS_LOGGING_ENABLED

Reply:
I can't find a way to inject anything in the IntegrationTestsSetup class to pass a logger to the containers

Reply:
protected override bool TestContainerLogging() => {
    
bool loggingEnabled = false;

if (bool.TryParse(Environment.GetEnvironmentVariable("FM_TESTCONTAINERS_LOGGING_ENABLED"), out var environmentLevelLoggingEnabled) && environmentLevelLoggingEnabled)
{
  loggingEnabled = true;
}

if (TestContext.Exists("FM_TESTCONTAINERS_LOGGING_ENABLED") && bool.TryParse(TestContext.Parameters["FM_TESTCONTAINERS_LOGGING_ENABLED"], out var nunitLevelLoggingEnabled) && nunitLevelLoggingEnabled))
{
    loggingEnabled = true
}


and then in your Builds

var builder =
new PostgreSqlBuilder().WithReuse(true);

if (LoggingEnabled())
{
  // see https://stackoverflow.com/a/78448295
 builder. WithLogging()
}
return builder.Build();

https://docs.nunit.org/articles/nunit/writing-tests/TestContext.html#testparameters

The reason I care about this is debugging some failures may require this level of logging especially on a build server.

Reply:
@jzabroski sure, I totally understand the need for logging, but WithLogging needs a ILogger which I can't get because I don't find a way to inject it or a logger factory into IntegrationTestsSetup. Or I need to use another way to start the containers, but I will need to make sure they are started only once...

Reply:
Got it. I will look into it today. I primarily use xunit day to day so when these things come up I need to think vs just do from muscle memory.